src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -71,6 +71,12 @@ public Guid(ReadOnlySpan<byte> b)
             _h = b[12];
             _i = b[13];
             _j = b[14];
+
+            [StackTraceHidden]
+            static void ThrowArgumentException()
```

Comment:
Should this also have `[DoesNotReturn]` ?

Reply:
It _could_, but it wouldn't help anything here.  [DoesNotReturn] was introduced as part of nullable reference types, and let's the compiler factor into its nullability flow analysis that the method won't return.  There's only one call site here and there aren't reference types involved.